chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
RE adding the Module.clock... If I call a function inside a withClock block and then inside that function it calls Module.clock... what clock are we talking about then?

Concrete example:

```
object ClockConnector {
  def apply(Clock ck) {
    ck := Module.clock
  }
}

... 

val FooModule extends Module {
  val io = new Bundle {val otherClock = Clock(INPUT) }
  val yetAnotherClock = Wire(Clock())

  withClockAndReset(io.otherClock) {
    ConnectClock(yetAnotherClock)  
  }
```

what is `yetAnotherClock` connected to?